edx_sga/static/js/src/edx_sga.js

```diff
@@ -142,6 +142,15 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                 });
             });
 
+            // Set up submission download links
+            var submissionDownloadedHtml = "<i class=\"fa fa-check-circle\" " +
```

Comment:
I think you don't need this. You can always have this in the DOM and just toggle hide / show, right?

Reply:
Generally speaking, it's a bad idea to have a template in JS code (if you don't really have to).
It's to satisfy the principle called `Separation of concerns`.
You want the template to be in one file where you can clearly see the structure, and achieve the dynamics using the JS code (scripts). But, changing the tree structure is a bad idea most of the time because you might end up with the bug that's hard to handle.
If the template is as static as possible, the only thing you have to worry about is hiding and showing certain parts of the tree.

Reply:
Yep, you have a good point, the show/hide approach makes sense.